macros/create_udfs.sql

```diff
@@ -0,0 +1,159 @@
+{% macro create_udfs() %}
+  {{ return(adapter.dispatch('create_udfs', 'snowplow_fractribution')()) }}
+{% endmacro %}
+
+
+{% macro snowflake__create_udfs() %}
+
+{{ config(
+  schema="derived"
+)}}
+  {% set trim_long_path %}
+  -- Returns the last path_lookback_steps channels in the path if path_lookback_steps > 0,
+  -- or the full path otherwise.
+  CREATE FUNCTION IF NOT EXISTS {{schema}}.TrimLongPath(path ARRAY, path_lookback_steps DOUBLE)
+  RETURNS ARRAY LANGUAGE JAVASCRIPT AS $$
+  if (PATH_LOOKBACK_STEPS > 0) {
+      return PATH.slice(Math.max(0, PATH.length - PATH_LOOKBACK_STEPS));
+    }
+    return PATH;
+  $$;
+  {% endset %}
+
+
+  -- Functions for applying transformations to path arrays.
+  -- unique: Identity transform.
+  --   E.g. [D, A, B, B, C, D, C, C] --> [D, A, B, B, C, D, C, C].
+  -- exposure: Collapse sequential repeats.
+  --   E.g. [D, A, B, B, C, D, C, C] --> [D, A, B, C, D, C].
+  -- first: Removes repeated events.
+  --   E.g. [D, A, B, B, C, D, C, C] --> [D, A, B, C].
+  -- frequency: Removes repeat events but tracks them with a count.
+  --   E.g. [D, A, B, B, C, D, C, C] --> [D(2), A(1), B(2), C(3)).
+
```

Comment:
Raised it already in the other PR, just noting that 'recency' might be missing, too.

Reply:
Thanks - I've removed that as it shouldn't have been there